magicblock-config/tests/read_config.rs

```diff
@@ -118,6 +118,7 @@ fn test_load_local_dev_with_programs_toml_envs_override() {
     env::set_var("VALIDATOR_FQDN", "magicblock.er.com");
     env::set_var("LEDGER_SIZE", "123123");
     env::set_var("LEDGER_RESUME_STRATEGY", "resume-only");
+    env::set_var("LEDGER_STARTING_SLOT", "1");
```

Comment:
The naming of this is a bit tricky once you make it part of a variant, but since it is only valid for certain variant I'd name it something as follows:

- `LEDGER_RESUME_STRATEGY_RESET_SLOT` which still makes it very clear which path to take in the toml config to define it as well.
You can then easily detect if it was supplied for a conflicting resume stragegy when trying to construct the variant.